File: src/p0088/cpp/solution.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
		vector<int> t(m);
		for(int i = 0; i < m; i++) {
			t[i] = nums1[i];
		}
		int i = 0, j = 0, k = 0;
		while(i < m && j < n) {
			nums1[k++] = t[i] > nums2[j] ? nums2[j++] : t[i++];
		}
		while(i < m) { nums1[k++] = t[i++]; }
		while(j < n) { nums1[k++] = nums2[j++]; }
    }
};

File: src/p0088/python/solution.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        t = [0] * (m + n)
        i, j, k = 0, 0, 0
        while i < m and j < n:
            if nums1[i] < nums2[j]:
                t[k] = nums1[i]
                i += 1
            else:
                t[k] = nums2[j]
                j += 1
            k += 1
        while i < m:
            t[k] = nums1[i]
            i += 1
            k += 1
        while j < n:
            t[k] = nums2[j]
            j += 1
            k += 1
        for i in range(m+n):
            nums1[i] = t[i]